fastapi_auth_api_key/dependencies.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from typing import Annotated
+
+from odoo import SUPERUSER_ID
+from odoo.api import Environment
+from odoo.exceptions import ValidationError
+
+from odoo.addons.auth_api_key.models.auth_api_key import AuthApiKey
+from odoo.addons.base.models.res_partner import Partner
+from odoo.addons.fastapi.dependencies import fastapi_endpoint, odoo_env
+from odoo.addons.fastapi.models.fastapi_endpoint import FastapiEndpoint
+
+from fastapi import Depends, status
+from fastapi.exceptions import HTTPException
+from fastapi.security import APIKeyHeader
+
+
+def authenticated_auth_api_key(
+    key: Annotated[str, Depends(APIKeyHeader(name="HTTP-API-KEY"))],
```

Comment:
is it a real desired changed ? to avoid breaking consumer I would recomand to use same header as on previous version ie; https://github.com/OCA/rest-framework/blob/16.0/base_rest_auth_api_key/apispec/rest_method_security_plugin.py#L17


```suggestion
    key: Annotated[str, Depends(APIKeyHeader(name="API-KEY"))],
```